pinot-core/src/main/java/org/apache/pinot/core/upsert/PartitionUpsertMetadataManager.java

```diff
@@ -83,34 +87,34 @@ public ThreadSafeMutableRoaringBitmap addSegment(String segmentName, Iterator<Re
           if (segmentName.equals(currentRecordLocation.getSegmentName())) {
             // The current record location has the same segment name
 
-            if (validDocIds == currentRecordLocation.getValidDocIds()) {
-              // The current record location is pointing to the new segment being loaded
-
-              // Update the record location when getting a newer timestamp
-              if (recordInfo._timestamp > currentRecordLocation.getTimestamp()) {
+            // Update the record location when the new timestamp is greater than or equal to the current timestamp.
+            // There are 2 scenarios:
+            //   1. The current record location is pointing to the new segment being loaded, where we want to update the
```

Comment:
Can you update this line of comments? It is a bit hard to understand. Do you mean:
"…, where we want to" =>. In this case, we want to update the record location when there is a tie…

also when this is a tie, how to you update the record location? Should it be determined by docID?